src/Paket.Core/Installation/RestoreProcess.fs

```diff
@@ -264,22 +265,49 @@ let createPaketPropsFile (cliTools:ResolvedPackage seq) restoreSuccess (fileInfo
             |> Seq.map (fun cliTool -> sprintf """        <DotNetCliToolReference Include="%O" Version="%O" />""" cliTool.Name cliTool.Version)
             |> fun xs -> String.Join(Environment.NewLine,xs)
             |> fun s -> "    <ItemGroup>" + Environment.NewLine + s + Environment.NewLine + "    </ItemGroup>"
-
-            
+    
+    let packagesParts =
+        if Seq.isEmpty packages then
+            ""
+        else
+            packages
+            |> Seq.map (fun ((groupName,packageName),_,_) -> 
+                let p = lockFile.Groups.[groupName].Resolution.[packageName]
+                let condition = Paket.Requirements.getExplicitRestriction p.Settings.FrameworkRestrictions      
+                p,condition)
+            |> Seq.groupBy snd
+            |> Seq.collect (fun (condition,packages) -> 
+                let condition = condition.ToMSBuildCondition()
+                let condition =
+                    if condition = "" || condition = "true" then "" else
+                    sprintf " AND (%s)" condition
+
+                let packageReferences =
+                    packages    
+                    |> Seq.collect (fun (p,_) ->
+                        [sprintf """        <PackageReference Include="%O">""" p.Name
+                         sprintf """            <Version>%O</Version>""" p.Version
+                        //"""            <PrivateAssets Condition=" ('%(PaketReferencesFileLinesInfo.AllPrivateAssets)' == 'true') Or ('$(PackAsTool)' == 'true') ">All</PrivateAssets>""" +
+                        //"""            <ExcludeAssets Condition="%(PaketReferencesFileLinesInfo.AllPrivateAssets) == 'exclude'">runtime</ExcludeAssets>""" +
+                        //"""            <Publish Condition=" '$(PackAsTool)' == 'true' ">true</Publish>""" +
+                         """        </PackageReference>"""])
+
+                [yield sprintf "    <ItemGroup Condition=\"($(DesignTimeBuild) == true)%s\">" condition
+                 yield! packageReferences
+                 yield "    </ItemGroup>"])
+            |> fun xs -> String.Join(Environment.NewLine,xs)
```

Comment:
Do you know that there is a [`String.concat`](https://msdn.microsoft.com/en-us/visualfsharpdocs/conceptual/string.concat-function-%5Bfsharp%5D?f=255&MSPPError=-2147217396) library routine?

Reply:
Yes, but what would that improve?